src/net/http.rs

```diff
@@ -170,14 +185,20 @@ async fn http_cache_get(context: &Context, url: &str) -> Result<Option<Response>
         }
     };
 
-    let expires = http_url_cache_expires(url, mimetype.as_deref());
+    let (expires, _stale) = http_url_cache_timestamps(url, mimetype.as_deref());
     let response = Response {
         blob,
         mimetype,
         encoding,
     };
 
-    // Update expiration timestamp.
+    // Update expiration timestamp
+    // to prevent deletion of the file still in use.
+    //
+    // We do not update stale timestamp here
```

Comment:
Still it makes sense to update `stale` to e.g. `now + 1*minute` to avoid revalidating very frequently (and have many parallel revalidation tasks) if revalidation fails or the HTTP request takes some time. The stale period is >= 1 hour, so 1 more minute won't be a problem